tm2/pkg/sdk/events.go

```diff
@@ -35,3 +38,63 @@ func (em *EventLogger) EmitEvents(events []Event) {
 // ----------------------------------------------------------------------------
 
 type Event = abci.Event
+
+func NewEvent(eventType string, pkgPath string, ident string, timestamp int64, attrs ...EventAttribute) Event {
+	return AttributedEvent{
+		Type:       eventType,
+		PkgPath:    pkgPath,
+		Identifier: ident,
+		Attributes: attrs,
+		Timestamp:  timestamp,
+	}
+}
+
+type AttributedEvent struct {
```

Comment:
How about renaming Attributes to Details and using this name instead?

```suggestion
type DetailedEvent struct {
```

Reply:
Or maybe includes Attributes or Details directly in the original Event as an optional field?

cc @jaekwon

Reply:
I also thought the name of this type was ambiguous. This name seems good too